File: app/src/main/java/com/example/pssmobile/retrofit/ApiClient.java
package com.example.pssmobile.retrofit;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {


    public static final String BASE_URL = "https://creator.zoho.com";
    private static final String BASE_URL_WRITER ="http://103.255.190.131/PSSApp/";
    private static RequestInterface loadInterface=null;
    private static Retrofit retrofit = null;
    private static Retrofit retrofitWriter = null;


    public static Retrofit getClient() {
        if (retrofit==null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }

    public static Retrofit getWriter() {
        if (retrofitWriter == null) {
            retrofitWriter = new Retrofit.Builder()
                    .baseUrl(BASE_URL_WRITER)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofitWriter;
    }

   /* public static RequestInterface loadInterface(){
        if (loadInterface ==null){
            loadInterface = ApiClient.getWriter().create(RequestInterface.class);
        }
        return loadInterface;
    }*/
}

File: app/src/main/java/com/example/pssmobile/retrofit/APIClient.kt
package com.example.pssmobile.retrofit

import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object APIClient {
    private var retrofit: Retrofit? = null

    fun getClient(baseUrl: String, zohoAuthToken: String): Retrofit? {

        val httpClient: OkHttpClient.Builder = OkHttpClient.Builder()
        httpClient.addInterceptor(object : Interceptor {
            override fun intercept(chain: Interceptor.Chain): Response {
                val request: Request = chain.request().newBuilder()
                        .addHeader("Authorization",zohoAuthToken)
                        .build()
                return chain.proceed(request)
            }
        })

        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                    .baseUrl(baseUrl)
                    .client(httpClient.build())
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }
        return retrofit
    }
}